honeybee_grasshopper_radiance/src/HB Extract HDR.py

```diff
@@ -0,0 +1,218 @@
+except ImportError as e:
+    raise ImportError('\nFailed to import honeybee_radiance:\n\t{}'.format(e))
+
+try:  # import ladybug_rhino dependencies
+    from ladybug_rhino.grasshopper import all_required_inputs, give_warning
+except ImportError as e:
+    raise ImportError('\nFailed to import ladybug_rhino:\n\t{}'.format(e))
+
+# check the Radiance date of the installed radiance
+try:  # import lbt_recipes dependencies
+    from lbt_recipes.version import check_radiance_date
+except ImportError as e:
+    raise ImportError('\nFailed to import lbt_recipes:\n\t{}'.format(e))
+check_radiance_date()
+
+
+def check_view_hdr(hdr_path):
+    """Check if the header of the HDR image contains a view (VIEW=).
+    
+    A ValueError is raised if the image does not contain a valid view.
+    A ValueError is raised if the view type is not -vta or -vth.
+    
+    Args:
+        hdr_path: The path to an HDR image file.
+    """
+    # set valid view = False. Will later be set to True if a valid view is found
+    valid_view = False
+    
+    # read hdr image and search for a valid view
+    with open(hdr_path, 'r') as hdr_file:
+        for lineCount, line in enumerate(hdr_file):
+            if lineCount < 200:
+                low_line = line.lower()
+                if not low_line.startswith('\t'):
+                    if low_line.startswith('view='):
+                        valid_view = True
+                        if line.find('-vth') > -1:
+                            projection = '-vth'
+                        elif line.find('-vta') > -1:
+                            projection = '-vta'
+                        hdr_view = View.from_string('hdr_view', line)
+            else:  # no need to check the rest of the document
+                break
+    if not valid_view:
+        raise ValueError(
+            'Connected _hdr image does not contain a valid view in the header.\n'
+            'Note that indented views in the header will be ignored by pinterp.')
+    if not hdr_view.type in ('a', 'h'):
+        msg = 'Expected view type -vta or -vth in _hdr. Got view type -vt{}.'
+        raise ValueError(msg.format(hdr_view.type))
+    return hdr_view
+
+
+def check_view_points(view, hdr_view):
+    """Check if view points of output view and input HDR are matching.
+    
+    A ValueError is raised if the view points are not matching.
+    
+    Args:
+        view: A Honeybee Radiance View to extract.
+        hdr_view: A Honeybee Radiance View from the input HDR.
+    """
+    if not view.position == hdr_view.position:
+        msg = 'View points of _view and _hdr are not matching.\n' \
+        'Got _view = {} and _hdr = {}.'
+        raise ValueError(msg.format(view.position, hdr_view.position))
+
+
+def check_resolution(hdr_path, resolution, view, hdr_view):
+    """Check the resolution of the output HDR as well as the input HDR.
+    
+    A warning is raised if the HDR image dimensions are not square. A warning is
```

Comment:
Am I correct that the Radiance command still succeeds even when the output resolution is larger than the input resolution? If so, this is good. Otherwise, it would be good to raise an error here instead of a warning.

Reply:
Yes, that is correct.